src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -118,8 +119,28 @@ export class IndexPatternsService {
       fields: ['title'],
       perPage: 10000,
     });
+
+    this.savedObjectsCache = await Promise.all(
+      this.savedObjectsCache.map(async (obj) => {
```

Comment:
we don't have memory usage tests can we verify there is no leak here.

Reply:
Updated try catch around `getDataSource()` that returns promises, to make sure it's handled. Besides, the amount data is Datasource meta data, which has very little fields, little amount of data returned

Reply:
nit. To follow the flow, maybe we can make 117 a local variable, e.g. `indexpatternCache`, and only assign the class attribute `savedObjectsCache` at 123. 

Given " NodeJs Reference counting: V8 keeps track of the number of references to an object. When the reference count of an object reaches zero, it becomes eligible for garbage collection." The local variable would be GC-ed after usage.